### PR TITLE
OMSDK-620: Migrate iOS reference app to use new OMID 1.3 API's and add audibility

### PR DESCRIPTION
**Pull request by jwhines**
Jira Link

https://theiab.atlassian.net/browse/OMSDK-620

This is an update to 1.3 API and added new view controller to test an audio ad.

There are a few things that still need to be updated:

1) Updated hosted verification script and creative.html (verification script is currently added as an asset and creative.html is being hosted on a test host. These will be removed once they are updated with correct scripts.)

2) Static lib is used within the project. Had a hard time wrapping in framework. Once 1.3 js is final, will use IABDemolab namespaced framework.

### CODE REVIEW

File: OM-Demo/ImageViewController.swift [hunk 1]
```diff
@@ -45,26 +45,24 @@ class ImageViewController: BaseAdUnitViewController {
...
     }
 
-    override func createAdSessionConfiguration() -> OMIDDemobuildAdSessionConfiguration {
+    override func createAdSessionConfiguration() -> OMIDAdSessionConfiguration {
         //Create ad session configuration
         do {
-            return try OMIDDemobuildAdSessionConfiguration(impressionOwner: .nativeOwner,
-                                                         videoEventsOwner: .noneOwner,
-                                                         isolateVerificationScripts: false)
+            return try OMIDAdSessionConfiguration(creativeType: .nativeDisplay, impressionType: .viewable, impressionOwner: .nativeOwner, mediaEventsOwner: .noneOwner, isolateVerificationScripts: true)
```
Comment (juo007) [inline OM-Demo/ImageViewController.swift]: Can you also include the display loaded event if the impressionOwner is native? 
Should come before the impression.

File: OM-Demo/VideoViewController.swift [hunk 2]
```diff
@@ -99,43 +104,47 @@ class VideoViewController: BaseAdUnitViewController {
...
+                                                                          verificationScriptURL: urlToMeasurementScript.absoluteString,
                                                                           parameters: parameters)
             else {
                 fatalError("Unable to instantiate session context: verification resource cannot be nil")
         }
 
         //Create native ad session context
         do {
-            return try OMIDDemobuildAdSessionContext(partner: partner, script: omidJSService, resources: [verificationResource], customReferenceIdentifier: nil)
+            return try OMIDAdSessionContext(partner: partner, script: omidJSService, resources: [verificationResource], customReferenceIdentifier: nil)
```
Comment (juo007) [inline OM-Demo/VideoViewController.swift]: Can you add contentUrl: nil here?

**Comment by jwhines** [comment]
Yes, I’ll merge this into Dev-1.3 with your requested changes. Im also getting the duplicate pings and late media events. Do you see the duplicates/lates on Android?

File: OM-Demo/ImageViewController.swift (hunk 1 above)
Comment (jwhines) [reply OM-Demo/ImageViewController.swift]: done.

File: OM-Demo/VideoViewController.swift (hunk 2 above)
Comment (jwhines) [reply OM-Demo/VideoViewController.swift]: done

**Comment by jwhines** [comment]
@juo007 I've changed the base to be merged into dev-1.3 upon approval.

File: OM-Demo/BridgingHeader.h [hunk 3]
```diff
@@ -0,0 +1,18 @@
+//
+//  BridgingHeader.h
+//  OM-Demo
+//
+//  Created by Justin Hines on 8/15/19.
+//  Copyright © 2019 Open Measurement Working Group. All rights reserved.
+//
+
+// TODO: Remove this file once lib is built as a framework
```
Comment (juo007) [inline OM-Demo/BridgingHeader.h]: Is this file necessary now?

File: OM-Demo/Constants.swift [hunk 4]
```diff
@@ -10,10 +10,28 @@ import Foundation
...
     
-    static let verificationScriptURL = "http://s3-us-west-2.amazonaws.com/omsdk-files/demo/creative/omid-validation-verification-script-v1.js"
+    static var audioAdURL: URL {
+        let url = Bundle.main.url(forResource: "audioAd", withExtension: "mp3")!
+        return url
+    }
+    
+    //TODO: update hosted omid-validation-verification-script-v1 to 1.3
+    
+//    static let verificationScriptURL = "http://www.pandora.com/static/ads/omsdk-v1_3/omid-validation-verification-script-v1.js"
```
Comment (juo007) [inline OM-Demo/Constants.swift]: can you remove this?

File: OM-Demo/BridgingHeader.h (hunk 3 above)
Comment (jwhines) [reply OM-Demo/BridgingHeader.h]: No, I can remove. Just using it for testing.

Comment (juo007) [reply OM-Demo/BridgingHeader.h]: cool thanks!

File: OM-Demo/Constants.swift (hunk 4 above)
Comment (juo007) [reply OM-Demo/Constants.swift]: *meaning just the comment

**Comment by juo007** [comment]
Tested with a framework built from OMSDK iOS dev-1.3 and confirmed all events in Charles. 
Removed framework and updated the README to point to downloading the .framework from the tools portal.

File: OM-Demo/BaseAdUnitViewController.swift [hunk 5]
```diff
@@ -199,26 +211,22 @@ class BaseAdUnitViewController: UIViewController {
...
+       
+        return OMIDDemoappSDK.shared.isActive
 
     }
 
-    private func createAdSession() -> OMIDDemobuildAdSession {
+    private func createAdSession() -> OMIDDemoappAdSession {
         //Partner name has to be unique to your integration
-        let partnerName = "Demobuild"
+        let partnerName = "Pandora"
```
Comment (garohussenjian) [inline OM-Demo/BaseAdUnitViewController.swift]: Should this be "Demoapp"?
